src/ShellClients/ShellClientsManager.vala

```diff
@@ -156,4 +190,50 @@ public class Gala.ShellClientsManager : Object {
 
         centered_windows[window] = new CenteredWindow (wm, window);
     }
+
+    //X11 only
+    private void parse_mutter_hints (Meta.Window window) requires (!Meta.Util.is_wayland_compositor ()) {
+        if (window.mutter_hints == null) {
+            return;
+        }
+
+        var mutter_hints = window.mutter_hints.split (":");
+        foreach (var mutter_hint in mutter_hints) {
+            var split = mutter_hint.split ("=");
+
+            if (split.length != 2) {
+                continue;
+            }
+
+            var key = split[0];
+            var val = split[1];
+
+            switch (key) {
+                case "anchor":
+                    set_anchor (window, int.parse (val));
```

Comment:
I would use `try_parse` just to be on the safe side

Reply:
Done :)